ppa/archive/forms.py

```diff
@@ -36,6 +79,35 @@ class SearchForm(forms.Form):
         'collections_exact': 'collections'
     }
 
+    def get_solr_sort_field(self, sort):
+        '''
+        Set solr sort fields for the query based on sort and query strings.
+
+        :return: tuple of strings following the format (template string,
+            solr sort string, solr fields string)
+        '''
+        solr_mapping = {
+            'relevance': 'score desc',
+            'pub_date_asc': 'pub_date asc',
+            'pub_date_desc': 'pub_date desc',
+            'title_asc': 'title_exact asc',
+            'title_desc': 'title_exact desc',
+            'author_asc': 'author_exact asc',
+            'author_desc': 'author_exact desc',
+        }
+        template_mapping = dict(self.SORT_CHOICES)
+        fields = '*'
+
+        if sort == 'relevance':
+            # relevance requires we include score field
+            fields = '*,score'
```

Comment:
I think the fields logic needs to be in the view, not the form (esp. since I've been thinking I'd like to pare down the fields to just the ones we actually need, instead of returning everything).  If it's any easier, we could just include score whenever there's a keyword query (not sure if there's a cost associated with calculating it?).

Reply:
I had pulled it to the form because it was dependent on the field set, but I think if we use the second option (and probably minimal cost) we can simplify the code significantly.